sparse/coo/indexing.py

```diff
@@ -450,3 +450,36 @@ def _join_adjacent_pairs(starts_old, stops_old):  # pragma: no cover
     stops.append(stops_old[-1])
 
     return starts, stops
+
+
+def _normalize_axis(axis, ndim):
+    """
+    Normalize negative axis indices to their positive counterpart for a given
+    number of dimensions.
+
+    Parameters
+    ----------
+    axis : Union[int, Iterable[int], None]
+        The axis indices.
+    ndim : int
+        Number of dimensions to normalize axis indices against.
+
+    Returns
+    -------
+    axis
+        The normalized axis indices.
+    """
+    if axis is None:
+        return None
+    if isinstance(axis, Iterable):
+        axis = tuple(ndim + a if a < 0 else a for a in axis)
+        for a in axis:
+            if a >= ndim or a < 0:
+                raise ValueError(
+                    "Invalid axis index %d for ndim=%d" % (a, ndim)
+                )
+        return axis
+    elif isinstance(axis, int):
+        return _normalize_axis([axis], ndim)[0]
```

Comment:
Can we flip the logic here? Make `int` the default and call that for `Iterable`?